tango/integrations/torch/train_config.py

```diff
@@ -184,12 +184,6 @@ def should_checkpoint_this_step(self, step: int) -> bool:
         assert self.train_steps is not None
         return ((step + 1) % self.checkpoint_every == 0) or step == self.train_steps - 1
 
-    def should_validate_this_step(self, step: int) -> bool:
```

Comment:
It's kind of shitty that all the `should_*` things are here, except for `should_validate`. But we don't have all the information we need here. We could leave the function here, and pass in the information of whether this is the last step of the epoch. It's just a little silly to say "if this parameter is `True` return `True` immediately and don't do anything."? But maybe that documents itself better.